lib/active_model/serializer/utils.rb

```diff
@@ -0,0 +1,25 @@
+module ActiveModel::Serializer::Utils
+  module_function
+
+  # Gives the existing lookup paths that are suffixes of the fully qualified path of `klass`.
+  # @param [Object] klass
+  # @return [Array<Symbol>]
+  def nested_lookup_paths(klass)
+    paths = klass.to_s.split('::').reverse.reduce([]) { |a, e| a + [[e] + Array(a.last)] }.reverse
```

Comment:
Would you mind helping me read this code?

1. `TweetSerializer::AuthorSerializer.to_s.split('::').reverse` #=> ['AuthorSerializer', 'TweetSerializer']`
2. `_.reduce([]) {|result, element| result + [[element] + Array(result.last)] }.reverse`
  1. `paths = []`
    1. `paths = [] + [['AuthorSerializer'] + Array([])  ] #=> [['AuthorSerializer']]`
    1. `paths = [['AuthorSerializer']] + [ ['TweetSerializer'] + Array(['AuthorSerializer']) ] #=> [ ["AuthorSerializer"], ["TweetSerializer", "AuthorSerializer"] ]`
  1. `paths.reverse #=> ["TweetSerializer", "AuthorSerializer"], ["AuthorSerializer"]]`
3. ` ["TweetSerializer", "AuthorSerializer"], ["AuthorSerializer"]].map! {|path| path.join('::') } #=> ["TweetSerializer::AuthorSerializer", "AuthorSerializer"]`
4. `[TweetSerializer::AuthorSerializer", "AuthorSerializer"].select! {|path| Object.const_defined?(path, false) }.map! {|path| Object.const_get(path) }.push(Object) #=> [`TweetSerializer::AuthorSerializer, AuthorSerializer, Object]`

Is that right?